### PR TITLE
Consolidate ansible directory hierarchy

### PR DESCRIPTION
**Pull request by portante**
We remove the extra level of `pbench/agent` under `agent/ansible`, while moving the playbooks and inventory into their own directory.

Before we do that, we remove the unused `ara` system information collection option.

And before that, we correct the references to `pbench-sysinfo-dump` in `CONFIG_DATA.md`.

### CODE REVIEW

**Review by webbnh — APPROVED** [review]
Other than the fact that I know next to nothing about Ansible, these changes look OK to me.

I do have one item for you to consider.

File: agent/ansible/playbooks/README [hunk 1]
```diff
@@ -16,7 +16,7 @@ N.B. Installation is incomplete until we install a key file so that
 config file so that it knows which server to use.
 
 The playbooks use variables defined in the inventory file (for an
-example inventory file, see =Inventory/pbench-agent.hosts.example=).
+example inventory file, see =./pbench-agent.hosts.example=).
```
[the remark this answers is not in the document]
Comment (portante) [reply agent/ansible/playbooks/README:19]: I think this is using some sort of `org` mode in Emacs.  I'll defer to @ndokos on this one.

[the remark this answers is not in the document]
Comment (ndokos) [reply agent/ansible/playbooks/README:19]: Yes - we probably should change it to Markdown and be done with it. In Org mode, =...= formats the stuff in-between the `=` signs in monospace.

**Review by ndokos — COMMENTED** [review]
Running `make test` in the `agent/ansible` directory produces  a bunch of errors. `ansible-lint` complains about using `latest` in package installs. It can be made to shut up about it by modifying the `ansible-lint.yml` file to add the line:
```
  - package-latest  # Package installs should not use latest.
 ```
to the `skip_list`.

Once past that, there are many error messages about the `check-readmes` script. The `ansible-test sanity` run thinks it's a python module and issues multiple complaints against it. Not sure how to exclude it from the sanity checking yet.

Here's the complete output I get:

```
$ make test
ansible-lint -c ansible-lint.yml
yaml-lint $(find . -type f -name '*.yml')
Checking the content of ["./ansible-lint.yml", "./galaxy.yml", "./meta/runtime.yml", "./playbooks/pbench_agent_config.yml", "./playbooks/pbench_agent_install.yml", "./playbooks/pbench_agent_tool_meister_firewall.yml", "./playbooks/pbench_repo_install.yml", "./roles/pbench_agent_config/defaults/main.yml", "./roles/pbench_agent_config/meta/main.yml", "./roles/pbench_agent_config/tasks/main.yml", "./roles/pbench_agent_files_install/meta/main.yml", "./roles/pbench_agent_files_install/tasks/main.yml", "./roles/pbench_agent_install/meta/main.yml", "./roles/pbench_agent_install/tasks/main.yml", "./roles/pbench_clean_yum_cache/meta/main.yml", "./roles/pbench_clean_yum_cache/tasks/main.yml", "./roles/pbench_firewall_open_ports/defaults/main.yml", "./roles/pbench_firewall_open_ports/meta/main.yml", "./roles/pbench_firewall_open_ports/tasks/main.yml", "./roles/pbench_repo_install/defaults/main.yml", "./roles/pbench_repo_install/meta/main.yml", "./roles/pbench_repo_install/tasks/main.yml", "./roles/pbench_repo_install/vars/main.yml"]
File : ./ansible-lint.yml, Syntax OK
File : ./galaxy.yml, Syntax OK
File : ./meta/runtime.yml, Syntax OK
File : ./playbooks/pbench_agent_config.yml, Syntax OK
File : ./playbooks/pbench_agent_install.yml, Syntax OK
File : ./playbooks/pbench_agent_tool_meister_firewall.yml, Syntax OK
File : ./playbooks/pbench_repo_install.yml, Syntax OK
File : ./roles/pbench_agent_config/defaults/main.yml, Syntax OK
File : ./roles/pbench_agent_config/meta/main.yml, Syntax OK
File : ./roles/pbench_agent_config/tasks/main.yml, Syntax OK
File : ./roles/pbench_agent_files_install/meta/main.yml, Syntax OK
File : ./roles/pbench_agent_files_install/tasks/main.yml, Syntax OK
File : ./roles/pbench_agent_install/meta/main.yml, Syntax OK
File : ./roles/pbench_agent_install/tasks/main.yml, Syntax OK
File : ./roles/pbench_clean_yum_cache/meta/main.yml, Syntax OK
File : ./roles/pbench_clean_yum_cache/tasks/main.yml, Syntax OK
File : ./roles/pbench_firewall_open_ports/defaults/main.yml, Syntax OK
File : ./roles/pbench_firewall_open_ports/meta/main.yml, Syntax OK
File : ./roles/pbench_firewall_open_ports/tasks/main.yml, Syntax OK
File : ./roles/pbench_repo_install/defaults/main.yml, Syntax OK
File : ./roles/pbench_repo_install/meta/main.yml, Syntax OK
File : ./roles/pbench_repo_install/tasks/main.yml, Syntax OK
File : ./roles/pbench_repo_install/vars/main.yml, Syntax OK
Done.
bin/check-readmes
ansible-galaxy collection build
Created collection for pbench.agent at /home/nick/src/pbench/github/pbench/worktrees/portante-2991/agent/ansible/pbench-agent-1.0.7.tar.gz
ansible-galaxy collection install pbench-agent-1.0.7.tar.gz -p ./collections
Starting galaxy collection install process
[WARNING]: The specified collections path '/home/nick/src/pbench/github/pbench/worktrees/portante-2991/agent/ansible/collections' is
not part of the configured Ansible collections paths '/home/nick/.ansible/collections:/usr/share/ansible/collections'. The installed
collection won't be picked up in an Ansible run.
Process install dependency map
Starting collection install process
Installing 'pbench.agent:1.0.7' to '/home/nick/src/pbench/github/pbench/worktrees/portante-2991/agent/ansible/collections/ansible_collections/pbench/agent'
pbench.agent:1.0.7 was installed successfully
git submodule update --init --recursive
cd collections/ansible_collections/pbench/agent; ansible-test sanity
/home/nick/src/pbench/github/pbench/worktrees/portante-2991/agent/ansible/collections/ansible_collections/pbench/agent
Running sanity test "action-plugin-docs"
Running sanity test "ansible-doc"
Running sanity test "changelog"
Running sanity test "compile" on Python 2.6
Running sanity test "compile" on Python 2.7
Running sanity test "compile" on Python 3.5
Running sanity test "compile" on Python 3.6
Running sanity test "compile" on Python 3.7
Running sanity test "compile" on Python 3.8
ERROR: Found 1 compile issue(s) on python 3.8 which need to be resolved:
ERROR: bin/check-readmes:4:7: SyntaxError: if [[ ! -f ./README.md ]] ;then
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/compile.html
Running sanity test "compile" on Python 3.9
ERROR: Found 1 compile issue(s) on python 3.9 which need to be resolved:
ERROR: bin/check-readmes:4:7: SyntaxError: if [[ ! -f ./README.md ]] ;then
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/compile.html
Running sanity test "compile" on Python 3.10
ERROR: Found 1 compile issue(s) on python 3.10 which need to be resolved:
ERROR: bin/check-readmes:4:7: SyntaxError: if [[ ! -f ./README.md ]] ;then
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/compile.html
Running sanity test "empty-init"
Running sanity test "future-import-boilerplate"
Running sanity test "ignores"
Running sanity test "import" on Python 2.6
Running sanity test "import" on Python 2.7
Running sanity test "import" on Python 3.5
Running sanity test "import" on Python 3.6
Running sanity test "import" on Python 3.7
Running sanity test "import" on Python 3.8
Running sanity test "import" on Python 3.9
Running sanity test "import" on Python 3.10
Running sanity test "line-endings"
Running sanity test "metaclass-boilerplate"
Running sanity test "no-assert"
Running sanity test "no-basestring"
Running sanity test "no-dict-iteritems"
Running sanity test "no-dict-iterkeys"
Running sanity test "no-dict-itervalues"
Running sanity test "no-get-exception"
Running sanity test "no-illegal-filenames"
Running sanity test "no-main-display"
Running sanity test "no-smart-quotes"
Running sanity test "no-unicode-literals"
Running sanity test "pep8"
Using cached pip 21.3.1 bootstrap script: /home/nick/.ansible/test/cache/get_pip_21_3_1.py
Collecting pip==21.3.1
  Using cached pip-21.3.1-py3-none-any.whl (1.7 MB)
Collecting setuptools==60.8.2
  Using cached setuptools-60.8.2-py3-none-any.whl (1.1 MB)
Collecting wheel==0.37.1
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel, setuptools, pip
Successfully installed pip-21.3.1 setuptools-60.8.2 wheel-0.37.1
Collecting pycodestyle==2.6.0
  Using cached pycodestyle-2.6.0-py2.py3-none-any.whl (41 kB)
Installing collected packages: pycodestyle
Successfully installed pycodestyle-2.6.0
ERROR: Found 31 pep8 issue(s) which need to be resolved:
ERROR: bin/check-readmes:3:7: E225: missing whitespace around operator
ERROR: bin/check-readmes:4:6: E201: whitespace after '['
ERROR: bin/check-readmes:4:13: E226: missing whitespace around arithmetic operator
ERROR: bin/check-readmes:4:23: E202: whitespace before ']'
ERROR: bin/check-readmes:4:26: E203: whitespace before ';'
ERROR: bin/check-readmes:4:27: E231: missing whitespace after ';'
ERROR: bin/check-readmes:4:27: E702: multiple statements on one line (semicolon)
ERROR: bin/check-readmes:5:5: E113: unexpected indentation
ERROR: bin/check-readmes:6:11: E225: missing whitespace around operator
ERROR: bin/check-readmes:9:6: E201: whitespace after '['
ERROR: bin/check-readmes:9:13: E226: missing whitespace around arithmetic operator
ERROR: bin/check-readmes:9:21: E226: missing whitespace around arithmetic operator
ERROR: bin/check-readmes:9:31: E202: whitespace before ']'
ERROR: bin/check-readmes:9:34: E203: whitespace before ';'
ERROR: bin/check-readmes:9:35: E231: missing whitespace after ';'
ERROR: bin/check-readmes:9:35: E702: multiple statements on one line (semicolon)
ERROR: bin/check-readmes:10:5: E113: unexpected indentation
ERROR: bin/check-readmes:11:11: E225: missing whitespace around operator
ERROR: bin/check-readmes:14:18: E226: missing whitespace around arithmetic operator
ERROR: bin/check-readmes:14:20: E203: whitespace before ';'
ERROR: bin/check-readmes:14:21: E231: missing whitespace after ';'
ERROR: bin/check-readmes:14:21: E702: multiple statements on one line (semicolon)
ERROR: bin/check-readmes:15:5: E113: unexpected indentation
ERROR: bin/check-readmes:15:10: E201: whitespace after '['
ERROR: bin/check-readmes:15:23: E226: missing whitespace around arithmetic operator
ERROR: bin/check-readmes:15:33: E202: whitespace before ']'
ERROR: bin/check-readmes:15:36: E203: whitespace before ';'
ERROR: bin/check-readmes:15:37: E231: missing whitespace after ';'
ERROR: bin/check-readmes:15:37: E702: multiple statements on one line (semicolon)
ERROR: bin/check-readmes:16:9: E113: unexpected indentation
ERROR: bin/check-readmes:17:15: E225: missing whitespace around operator
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/pep8.html
Running sanity test "pslint"
Running sanity test "pylint"
Using cached pip 21.3.1 bootstrap script: /home/nick/.ansible/test/cache/get_pip_21_3_1.py
Collecting pip==21.3.1
  Using cached pip-21.3.1-py3-none-any.whl (1.7 MB)
Collecting setuptools==60.8.2
  Using cached setuptools-60.8.2-py3-none-any.whl (1.1 MB)
Collecting wheel==0.37.1
  Using cached wheel-0.37.1-py2.py3-none-any.whl (35 kB)
Installing collected packages: wheel, setuptools, pip
Successfully installed pip-21.3.1 setuptools-60.8.2 wheel-0.37.1
Collecting pylint==2.9.3
  Downloading pylint-2.9.3-py3-none-any.whl (372 kB)
     |████████████████████████████████| 372 kB 6.3 MB/s            
Collecting pyyaml==5.4.1
  Using cached PyYAML-5.4.1-cp39-cp39-manylinux1_x86_64.whl (630 kB)
Collecting astroid==2.6.6
  Downloading astroid-2.6.6-py3-none-any.whl (231 kB)
     |████████████████████████████████| 231 kB 8.9 MB/s            
Collecting isort==5.9.3
  Downloading isort-5.9.3-py3-none-any.whl (106 kB)
     |████████████████████████████████| 106 kB 11.0 MB/s            
Collecting lazy-object-proxy==1.6.0
  Downloading lazy_object_proxy-1.6.0-cp39-cp39-manylinux1_x86_64.whl (57 kB)
     |████████████████████████████████| 57 kB 6.2 MB/s             
Collecting mccabe==0.6.1
  Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Collecting toml==0.10.2
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting wrapt==1.12.1
  Downloading wrapt-1.12.1.tar.gz (27 kB)
  Preparing metadata (setup.py) ... done
Building wheels for collected packages: wrapt
  Building wheel for wrapt (setup.py) ... done
  Created wheel for wrapt: filename=wrapt-1.12.1-cp39-cp39-linux_x86_64.whl size=71657 sha256=ec28c1ae67b98a5fbe9db22e05e38e072bde4430308c4943aeb4cb94a666f139
  Stored in directory: /home/nick/.cache/pip/wheels/98/23/68/efe259aaca055e93b08e74fbe512819c69a2155c11ba3c0f10
Successfully built wrapt
Installing collected packages: wrapt, lazy-object-proxy, toml, mccabe, isort, astroid, pyyaml, pylint
Successfully installed astroid-2.6.6 isort-5.9.3 lazy-object-proxy-1.6.0 mccabe-0.6.1 pylint-2.9.3 pyyaml-5.4.1 toml-0.10.2 wrapt-1.12.1
ERROR: Found 1 pylint issue(s) which need to be resolved:
ERROR: bin/check-readmes:4:7: syntax-error: invalid syntax (<unknown>, line 4)
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/pylint.html
Running sanity test "replace-urlopen"
Running sanity test "runtime-metadata"
Running sanity test "shebang"
ERROR: Found 2 shebang issue(s) which need to be resolved:
ERROR: bin/check-readmes:1:1: unexpected non-module shebang: b'#! /bin/bash'
ERROR: bin/yamlconf:1:1: unexpected non-module shebang: b'#! /usr/bin/env python'
See documentation for help: https://docs.ansible.com/ansible-core/2.12/dev_guide/testing/sanity/shebang.html
Running sanity test "shellcheck"
Running sanity test "symlinks"
Running sanity test "use-argspec-type-path"
Running sanity test "use-compat-six"
Running sanity test "validate-modules"
Running sanity test "yamllint"
ERROR: The 6 sanity test(s) listed below (out of 46) failed. See error output above for details.
compile --python 3.8
compile --python 3.9
compile --python 3.10
pep8
pylint
shebang
make: *** [Makefile:26: test] Error 1
```

**Comment by vishalvvr** [comment]
> 
> Thanks, @vishalvvr! AFAICT, adding a `tests/sanity/ignore-2.12.txt` with the above contents eliminates the errors on my setup. A generic `ignore.txt` does not do it: shouldn't it?

generic `ignore.txt` is not supported in ansible collection.
As doc says 
[Ansible Collections](https://docs.ansible.com/ansible/latest/dev_guide/testing/sanity/ignores.html#ansible-collections)
``` 
Since sanity tests change between Ansible releases, a separate ignore file is needed for each Ansible major release.
The filename is tests/sanity/ignore-X.Y.txt where X.Y is the Ansible release being used to test the collection.
Maintaining a separate file for each Ansible release allows a collection to pass tests for multiple versions of Ansible.
```

**Comment by ndokos** [comment]
OK - is `ignore-2.12.txt` enough for now? Do we need more that that?

**Comment by vishalvvr** [comment]
> OK - is `ignore-2.12.txt` enough for now? Do we need more than that?

F36 has `ansible-core-2.12` and f37 & f38 has `ansible-core-2.13` [packaged](https://src.fedoraproject.org/rpms/ansible-core)
also [pypi](https://pypi.org/project/ansible-core/) has `ansible-core-2.13` packaged.

So better to have one more ignore file for 2.13
Again it all depends on the test machine and what ansible-core version it has.

**Comment by portante** [comment]
I'll post a proposed solution.

**Review by portante — COMMENTED** [review]
With this latest set of changes the `make test` target works cleanly in the `agent/ansible` directory.

File: agent/ansible/Makefile [hunk 2]
```diff
@@ -14,10 +13,12 @@ colldir = collections/ansible_collections/${namespace}/${collection}
 .DEFAULT_GOAL := test
 
 lint:
-	make -C pbench/agent
+	ansible-lint -c ansible-lint.yml
```
Comment (ndokos) [inline agent/ansible/Makefile:16]: The`ansible-lint.yml` file needs the addition

```
  - package-latest  # Package installs should not use latest.
```
to stop complaining about `latest`.

**Review by ndokos — CHANGES_REQUESTED** [review]

**Comment by webbnh** [comment]
> I'll post a proposed solution.

We're sure that we want to ignore the errors...as opposed, say, to fixing them?

**Review by webbnh — APPROVED** [review]
Looks OK to me, modulo my previous open conversation.

**Review by ndokos — APPROVED** [review]

**Review by webbnh — APPROVED** [review]
Looks OK to me, modulo [my previous open conversation](https://github.com/distributed-system-analysis/pbench/pull/2991#discussion_r960179870)...which, since Nick has approved, I guess I'll just close.

**Review by vishalvvr — APPROVED** [review]

**Pull request merged**